[jIicho] - 클릭하는 탭에 따라 컴포넌트가 달라지는 기능 구현

jIicho opened the pull request:
## 완료 작업 목록
  1. NewsDashboard 컴포넌트를 좀 더 작게 나누기 (styled.components 적용)
  2. 전체 언론사와 내가 구독한 언론사 버튼 클릭 시 컴포넌트 변경 
     - 전체 언론사 기본 displayMode는 GridView 적용
     - 내가 구독한 언론사 기본 displayMode는 ListView 적용
  3. mockData.json 파일 내에 Subscribed 항목을 추가하여 `전체 언론사`와`내가 구독한 언론사` 렌더링 되는 정보가 달라지게끔 처리 
     - ❓ AllPress 와 Subscribed key를 따로 만드는 것이 아니라 
          AllPress 데이터 객체 내에 subscribed를 추가하는 것이 나을까??? ❓ 

## 주요 고민과 해결과정
[고민과 해결방법](https://www.notion.so/1c6c433e887e8040916fd1f1b7ed9854?pvs=4)

jIicho committed:
chore: 파일 명 변경

jIicho committed:
feat: gridList border 중복 없이 만들기

jIicho committed:
fix: 이전 코드에서 비어있는 셀에 id값이 존재하지 않아서 발생한 오류 해결

jIicho committed:
refactor: newsDashboard 영역 컴포넌트 분리 및 일반 CSS 대신 styled-components로 변경

jIicho committed:
feat: useState를 사용하여 클릭 시 해당 스타일 적용되게끔 처리

jIicho committed:
feat: 부모 컴포넌트에서 상태 관리를 하기 위해 코드 수정

1. 이전 MidiaTabs.jsx, DisplayMode.jsx에서 상태 관리를 함
2. NewsDashboard.jsx에서 NewsDashboardContentsArea를 불러오기 때문에 NewsDashboard에서 상태 관리를 할 수 있도록 함
3. props를 전달하여 그 값을 토대로 처리됨

jIicho committed:
feat: activeTab에 따라 displayMode 상태 변화 시키기

1. 전체 언론사 탭 클릭 시 GridView가 기본으로 활성화되게 한다
2. 내가 구독한 언론사 탭 클릭 시 ListView가 기본으로 활성화 되게 한다

jIicho committed:
feat: 내가 구독한 언론사 클릭 시 ListView 컴포넌트가 보이게끔 처리

jIicho committed:
feat: props를 전달하여 상황에 따라 컴포넌트를 변경

1. 부모 컴포넌트 DashboardContent 작성
2. activeTab, displayMode를 props를 전달
3. 상황에 따라 GridView, ListView가 뜨게끔 처리

jIicho committed:
feat: GridView 내의 언론사를 AllPress 와 Subscribed 정보 처리

1. 이전의 mockData.json파일 내에 Subscribed 내용을 추가
2. activeTab에 따라 GridView를 달리 하기 위한 코드 작성
3. activeTab이 AllPress일 경우 : AllPress 정보를 담아 GridView 작성
4. activeTab이 Subscribed일 경우 : Subscribed 정보를 담아 GridView 작성

jIicho committed:
refactor: NewsDashboardContent 리팩토링

1. displayMode가 GridView인지 확인하는 로직 분리
2. title 변수를 선언하여 activeTab에 따라 작성할 문자열 정의
3. DashboardContentArea 컴포넌트 { }내부의 조건문 간결하게 수정

jIicho committed:
feat: ListView 전체 구조 잡기

crongro commented on public/mockData.json:
```diff
@@ -1,82 +1,94 @@
-  }
-]
+{
+  "AllPress": [
```
전에도 말했지만 이럴거면 그냥 반복문으로 표현해도 될 거 같아요.

crongro commented on src/components/newsDashboard/NewsDashboard.jsx:
```diff
@@ -0,0 +1,27 @@
+  const [displayMode, setDisplayMode] = useState("GridView");
+
+  useEffect(() => {
+    if (activeTab === "Subscribed") setDisplayMode("ListView");
```
이 코드는 useeffect에 굳이 둘 필요가 없을거 같아요. 
잘못하면 리렌더링이 두 번 일어날수도 있고요.

crongro commented on src/components/newsDashboard/contentArea/GridView.jsx:
```diff
@@ -0,0 +1,52 @@
+          : setPressData(data.Subscribed);
+      })
+      .catch((error) => console.error("Error:", error));
+  }, [activeTab]);
```
activeTab 이 바뀔때마다 실행되는 코드군요.
dependency에 대해서 잘 알아두세요.

crongro commented on src/components/newsDashboard/contentArea/GridView.jsx:
```diff
@@ -0,0 +1,52 @@
+    <GridContent>
+      {filledCells.map((press, index) => (
+        <Cell
+          key={press.id ? `${press.id}` : `empty-${index}`}
```
key={press.id ?? `empty-${index}`}
 

nullish coalescing operator

crongro reviewed (commented):
컴포넌트 개발 진전이 보이네요.
지금 컴포넌트 구조가 괜찮으니 계속 진행해보세요.

jIicho committed:
refactor: NewsDashboard.jsx 파일 피드백 적용

useEffect()코드로 인해 잘못하면 리렌더링이 두번 발생할 수 있음
1. useEffect() 코드 주석처리
2. NewsDashboardHeader.jsx파일에서 MediaTabs 컴포넌트에 setDisplayMode를 props로 전달
3. MediaTabs.jsx파일에서 setDisplayMode props를 받고 handleClick 함수에 조건문 추가

crongro commented:
"일부 파일은 소문자로 시작하네요. 모두 대문자로 통일할 필요가 있어요"
(by AI)

jIicho committed:
refactor: GridView.jsx 피드백 적용

nullish coalescing operator 사용

jIicho committed:
feat: 리스트 보기 UI 구현

jIicho committed:
refactor: 컴포넌트 파일명 정리

jIicho committed:
feat & refactor: RollingArea.jsx

1. 이전 기본 CSS적용 -> styled.components 으로 수정
2. 컴포넌트 a태그로 변경 가능하게 수정
3. &:hover 적용

jIicho committed:
chore: 컴포넌트 이름 수정

crongro commented on src/components/newsDashboard/contentArea/listView/SubNews.jsx:
```diff
@@ -0,0 +1,44 @@
+  return (
+    <Container>
+      <NewsTitle>
+        <span>"위스키 사려고 이틀 전부터 줄 섰어요"</span>
+        <span>'방시혁 제국'이냐 '카카오 왕국'이냐…K엔터 누가 거머쥘까</span>
+        <span>사용후핵연료 저장시설 포화…이대론 7년 뒤 원전 멈춘다</span>
+        <span>
+          [단독] 원희룡 "해외건설 근로자 소득공제 월 500만원으로 상향할 것"
+        </span>
+        <span>태평양에는 우영우의 고래만 있는게 아니었다 [로비의 그림]</span>
+        <span>LG엔솔, 폴란드 자동차산업협회 가입…“유럽서 목소리 키운다”</span>
```
테스트 목적이 아니면, 이런 데이터를 배열이나 객체로 보관해두고 사용~

jIicho committed:
feat: Express서버 프로젝트에서 크롤링하고 저장한 데이터를 요청 후 불러와 셀 체우기

jIicho committed:
feat: 기획서 그리드 형식에 맞게 처리되게끔 구현

1. 한 화면에 24개의 언론사만 보이게끔 처리
2. 첫 페이지에는 왼쪽 버튼 비활성화
3. 마지막 페이지에는 오른쪽 버튼 비활성화
4. 버튼을 포함한 GridContent 레이아웃 잡기
